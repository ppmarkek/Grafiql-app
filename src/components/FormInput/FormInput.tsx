import {
  FormControl,
  FormHelperText,
  InputLabel,
  OutlinedInput,
} from '@mui/material';
import { FormikProps } from 'formik';

export interface Values {
  name: string;
  email: string;
  password: string;
  passwordConfirm: string;
}

type FormInputType = {
  title: 'name' | 'email' | 'password' | 'passwordConfirm';
  type: 'text' | 'email' | 'password';
  text: string;
  formik: FormikProps<Values>;
};
const FormInput = ({ title, type, text, formik }: FormInputType) => {
  return (
    <FormControl error={!!formik.errors[title]} variant="outlined">
      <InputLabel htmlFor={title}>{text}</InputLabel>
      <OutlinedInput
        label={text}
        id={title}
        type={type}
        onChange={formik.handleChange}
        value={formik.values[title]}
        aria-describedby="component-error-text"
      />
      {formik.errors[title] && (
        <FormHelperText id="component-error-text">
          {formik.errors[title]}
        </FormHelperText>
      )}
    </FormControl>
  );
};

export default FormInput;
