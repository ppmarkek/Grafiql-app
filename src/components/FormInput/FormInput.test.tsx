import React from 'react';
import { FormikProps } from 'formik';
import FormInput, { Values } from './FormInput';
import renderer from 'react-test-renderer';

describe('FormInput', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('should render without errors', () => {
    const tree = renderer
      .create(
        <FormInput
          title="name"
          text="text"
          type="text"
          formik={
            {
              values: {
                name: '',
              },
              errors: {},
              handleChange: jest.fn(),
            } as unknown as FormikProps<Values>
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render with error', () => {
    const tree = renderer
      .create(
        <FormInput
          title="name"
          text="text"
          type="text"
          formik={
            {
              values: {
                name: '',
              },
              errors: { name: 'Required field' },
              handleChange: jest.fn(),
            } as unknown as FormikProps<Values>
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render with unrelated error', () => {
    const tree = renderer
      .create(
        <FormInput
          title="name"
          text="text"
          type="text"
          formik={
            {
              values: {
                name: '',
              },
              errors: { differentField: 'Required field' },
              handleChange: jest.fn(),
            } as unknown as FormikProps<Values>
          }
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
