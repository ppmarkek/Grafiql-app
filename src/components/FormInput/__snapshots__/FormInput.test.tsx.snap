// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render with error 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="name"
  >
    text
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-describedby="component-error-text"
      aria-invalid={true}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-yjsfm1"
      >
        <span>
          text
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    className="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    id="component-error-text"
  >
    Required field
  </p>
</div>
`;

exports[`FormInput should render with unrelated error 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="name"
  >
    text
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-describedby="component-error-text"
      aria-invalid={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-yjsfm1"
      >
        <span>
          text
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`FormInput should render without errors 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="name"
  >
    text
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-describedby="component-error-text"
      aria-invalid={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-yjsfm1"
      >
        <span>
          text
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
